In [110]:
print("Выведем некоторые из первых чисел N, при которых выполняется \n \
условие, что орлов на 10% больше чем решек и сохраняется целочисленность")
print("N:", "Орлов из них:", sep = '\t')

possiblens = dict()

for i in range(1,11000):
    if round(i * 1.1, 5) % 1 == 0:
        if i < 200:
            print(int(i+i*1.1), int(i*1.1), sep = '\t')
        possiblens[int(i+i*1.1)] = int(i*1.1)
        

Выведем некоторые из первых чисел N, при которых выполняется 
 условие, что орлов на 10% больше чем решек и сохраняется целочисленность
N:	Орлов из них:
21	11
42	22
63	33
84	44
105	55
126	66
147	77
168	88
189	99
210	110
231	121
252	132
273	143
294	154
315	165
336	176
357	187
378	198
399	209


### Воспользуемся  тестом для проверки параметра биномиального распределения $p_o= 0.5$ при $\alpha=0.5$
### $H_0: p_o <= 0.5$
### $H_1: p_o > 0.5$

In [107]:
import scipy.stats as stats

for i,j in possiblens.items():  #Проходимся по количеству экспериментов и успешным результатам (успешным считаестя "Орёл")
    if stats.binom_test(x=j, n=i, p=0.5, alternative='greater') < 0.05:  #Функция возвращает pv, сравниваем с alpha
        print(i,j)
        break #Когда pv оказывается меньше alpha останавливаем цикл и выводим нужный N

1239 649


### При $N = 1239$ $pvalue < \alpha$, следовательно отвергается гипотеза $H_0$ и можно сделать вывод о том, что монетка "нечестная", вероятность выпадения "Орла" оказалась выше 0.5